### ISSUE TITLE
GDScript is not able to infer class member types.

### ISSUE THREAD

**Issue by AndreaCatania** [godotengine/godot#43503]
**Godot version:**
f73374629dd57d3eea6a960859ca223fa2d95fc4

**OS/device including version:**
Fedora 32

**Issue description:**
The GDS analyzer is not able to infer the member type, when the variable is initialized to `null` and assigned after it's used:

![Screenshot from 2020-11-13 16-14-29](https://user-images.githubusercontent.com/8342599/99087847-88e17d80-25cb-11eb-894a-f32e1d3a655c.png)

If you move the `_ready` function above the `_process` function, the error disappear.

**Script:**
```gdscript
extends Node

var test_var = null


func _process(delta):
	print(test_var.x)


func _ready():
	test_var = Vector3()
```

**Comment by GodotDeveloper** [comment]
You can't use an 'x' value of a variable if you have the variable set to null (NiL). To fix the issue, just set the variable to a Vector value or put the "func _ready(): text_var = Vector3()" above the "func _process(delta): print(test_var.x)". You should also pay attention and read what the errors say, they can help at least a little with most of your problems.

TLDR: Just put the "ready" function above the "process" function, or, just set the variable to a Vector value instead of null.

**Comment by lyuma** [comment]
This issue seems very similar to #49183 which is a bug because `null` is a valid value for `: Object`.

The one big difference with this issue is that a primitive type like Vector3 can never be compared to null, so null never makes sense as a value for a Vector3.

If you really wish to insist on assigning null to a Vector3 type, GDScript 2.0 offers a means to escape from the type safety by declaring a variable as `: Variant`.

I'd be in favor of closing this issue: my assessment is that this is a feature, not a bug: type checking is doing its job of preventing you from mixing `null` and `Vector3` which are incompatible types.

That said, my guess is that lots of users will probably run into this problem when upgrading projects from Godot 3.x - perhaps some good documentation or error messages could help out.
